packages/graphql/src/services/learningElements.ts

```diff
@@ -194,161 +193,155 @@ export async function respondToQuestionInstance(
+      let updatedResults:
+        | {
+            choices?: Record<string, number>
+          }
+        | Record<string, number> = {}
+
+      switch (questionData.type) {
+        case QuestionType.SC:
+        case QuestionType.MC:
+        case QuestionType.KPRIM: {
+          updatedResults.choices = response.choices!.reduce(
+            (acc, ix) => ({
+              ...acc,
+              [ix]: acc[ix] + 1,
+            }),
+            results.choices as Record<string, number>
+          )
+          break
         }
 
-        const parsedValue = parseFloat(response.value)
-        if (
-          isNaN(parsedValue) ||
-          (typeof questionData.options.restrictions!.min === 'number' &&
-            parsedValue < questionData.options.restrictions!.min) ||
-          (typeof questionData.options.restrictions!.max === 'number' &&
-            parsedValue > questionData.options.restrictions!.max)
-        ) {
-          return {}
-        }
+        case QuestionType.NUMERICAL: {
+          if (
+            typeof response.value === 'undefined' ||
+            response.value === null ||
+            response.value === ''
+          ) {
+            return {}
+          }
 
-        const value = String(parsedValue)
+          const parsedValue = parseFloat(response.value)
+          if (
+            isNaN(parsedValue) ||
+            (typeof questionData.options.restrictions!.min === 'number' &&
+              parsedValue < questionData.options.restrictions!.min) ||
+            (typeof questionData.options.restrictions!.max === 'number' &&
+              parsedValue > questionData.options.restrictions!.max)
+          ) {
+            return {}
+          }
 
-        if (Object.keys(results).includes(value)) {
-          updatedResults = {
-            ...results,
-            [value]: (results as NumericalQuestionResults)[value] + 1,
+          const value = String(parsedValue)
+
+          if (Object.keys(results).includes(value)) {
+            updatedResults = {
+              ...results,
+              [value]: results[value] + 1,
+            }
+          } else {
+            updatedResults = { ...results, [value]: 1 }
           }
-        } else {
-          updatedResults = { ...results, [value]: 1 }
+          break
         }
-        break
-      }
 
-      case QuestionType.FREE_TEXT: {
-        if (
-          typeof response.value === 'undefined' ||
-          response.value === null ||
-          response.value === '' ||
-          (typeof questionData.options.restrictions!.maxLength === 'number' &&
-            response.value.length >
-              questionData.options.restrictions!.maxLength)
-        ) {
-          return {}
-        }
+        case QuestionType.FREE_TEXT: {
+          if (
+            typeof response.value === 'undefined' ||
+            response.value === null ||
+            response.value === '' ||
+            (typeof questionData.options.restrictions!.maxLength === 'number' &&
```

Comment:
Same here as above for numerical question type as TS again correctly recognizes `QuestionOptionsFreeText`

Reply:
true, fixed with ?

Reply:
types should be consistent then, yes :)